File: qt/src/config/TeleTuConfigParser.cpp
/*
 * Copyright 2013 Rui Araújo, Luís Fonseca
 *
 * This file is part of Router Keygen.
 *
 * Router Keygen is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * Router Keygen is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with Router Keygen.  If not, see <http://www.gnu.org/licenses/>.
 */
#include "TeleTuConfigParser.h"
#include <iostream>
#include <stdio.h>
#include <QFile>
#include <QTextStream>
#include <QStringList>

QMap<QString ,QVector<TeleTuMagicInfo *> *> * TeleTuConfigParser::readFile(const QString &fileName) {
    QFile file(fileName);
    if(!file.open(QIODevice::ReadOnly)) {
        return NULL;
    }
    QTextStream in(&file);
    int range[2];
    QMap<QString ,QVector<TeleTuMagicInfo *> *> * supportedTeleTus = new QMap<QString ,QVector<TeleTuMagicInfo *> *>();
    while(!in.atEnd()) {
        QString line = in.readLine();
        QStringList infos = line.split(" ");
        QString name = infos[0];
        QVector<TeleTuMagicInfo*> * supported = supportedTeleTus
                ->value(name);
        if (supported == NULL) {
            supported = new QVector<TeleTuMagicInfo*>();
            supportedTeleTus->insert(name, supported);
        }
        range[0] = infos[1].toInt(NULL,16); // from
        range[1] = infos[2].toInt(NULL,16); // to
        QString serial = infos[3];
        int base = infos[4].toInt(NULL,16);
        int divider = infos[5].toInt(NULL,10);
        supported->append(new TeleTuMagicInfo(range, serial, base, divider));
    }
    file.close();
    return supportedTeleTus;
}

File: android/routerKeygen/src/main/java/org/exobel/routerkeygen/config/TeleTuConfigParser.java
/*
 * Copyright 2012 Rui Araújo, Luís Fonseca
 *
 * This file is part of Router Keygen.
 *
 * Router Keygen is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * Router Keygen is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with Router Keygen.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.exobel.routerkeygen.config;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class TeleTuConfigParser {

	public static Map<String, ArrayList<TeleTuMagicInfo>> parse(InputStream in) {
		Map<String, ArrayList<TeleTuMagicInfo>> supportedTeleTu = new HashMap<String, ArrayList<TeleTuMagicInfo>>();
		final BufferedReader bufferedInput = new BufferedReader(
				new InputStreamReader(in));
		try {
			String line;
			while ((line = bufferedInput.readLine()) != null) {
				final String[] infos = line.split(" ");
				final String name = infos[0];
				ArrayList<TeleTuMagicInfo> supported = supportedTeleTu
						.get(name);
				if (supported == null) {
					supported = new ArrayList<TeleTuMagicInfo>(5);
					supportedTeleTu.put(name, supported);
				}
				int[] range = new int[2];
				range[0] = Integer.parseInt(infos[1], 16); // from
				range[1] = Integer.parseInt(infos[2], 16); // to
				final String serial = infos[3];
				final int base = Integer.parseInt(infos[4], 16);
				final int divider = Integer.parseInt(infos[5]);
				supported
						.add(new TeleTuMagicInfo(range, serial, base, divider));
			}
		} catch (Exception e) {
			e.printStackTrace();
		}
		return supportedTeleTu;
	}

}
